### PR TITLE
Add fn for basic plotting capabilities

### PR DESCRIPTION
**Pull request by Eleonore9**
Addresses issue #15 and fix issue #22 

* [x] Separated the creation of the plot and the rendering of the plot so I can add a test.
* [x] Handle the plotting of negative values
* [x] Add line on the 0 y-axis when there are negative values

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -9,8 +9,7 @@
                  [prismatic/schema "1.1.3"]
                  [org.clojure/data.csv "0.1.3"]
                  [schema-contrib "0.1.3"]
-                 [instaparse "1.4.3"]]
-  :main ^:skip-aot kixi.lapita
-  :target-path "target/%s"
+                 [instaparse "1.4.3"]
```
Comment (acron0) [inline project.clj:12]: Why instaparse?

File: src/kixi/lapita/plot.clj [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     :layout     viz/svg-bar-plot
+     :interleave num
+     :bar-width  width
+     :offset     idx}))
+
+(defn viz-spec
+  [x-data y-data plot-width plot-height]
+  {:x-axis (viz/linear-axis
+            {:domain [(let [min-x (reduce min x-data)]
+                        (if (= min-x 0) 0 (dec min-x)))
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: What about if you want to include numbers < 0 ?

File: src/kixi/lapita/plot.clj [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                      (inc (reduce max x-data))]
+             :range  [50 (- plot-width 20)]
+             :major  1
+             :pos    (- plot-height 40)
+             :label  (viz/default-svg-label int)})
+   :y-axis (viz/linear-axis
+            {:domain      [0 (+ 5 (reduce max y-data))]
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: What about if you want to include numbers < 0 ?

File: src/kixi/lapita/plot.clj [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+         bar-width (calc-width-of-bar x-data)
+         color-plot (or plot-color "#3D325A")
+         width-plot (or plot-width 600)
+         height-plot (or plot-height 320)]
+     (-> (viz-spec x-data y-data width-plot height-plot)
+         (assoc :data [((bar-spec all-data 2 bar-width) 0 color-plot)])
+         (viz/svg-plot2d-cartesian)
+         (export-viz output-path width-plot height-plot)))))
+
+(comment
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: Move to a test namespace pls

**Review by acron0 — CHANGES_REQUESTED** [review]

File: project.clj (hunk 1 above)
Comment (Eleonore9) [reply project.clj:12]: I think I need it for schema-contrib to work.

File: src/kixi/lapita/plot.clj (hunk 3 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: Good point, I need to test that!

File: src/kixi/lapita/plot.clj (hunk 2 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: Good point, I need to test that!

File: src/kixi/lapita/plot.clj (hunk 4 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: ok

File: src/kixi/lapita/plot.clj [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+             :pos         50
+             :label-dist  15
+             :label-style {:text-anchor "end"}})
+   :grid   {:minor-y true}})
+
+(defn plot-bar-chart
+  ([ds x-axis y-axis output-path]
+   (plot-bar-chart ds x-axis y-axis output-path {:plot-color "#3D325A"
+                                                 :plot-width 600
+                                                 :plot-height 320}))
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: No need to include defaults in here if they occur in the other arity

**Review by acron0 — CHANGES_REQUESTED** [review]

File: src/kixi/lapita/plot.clj (hunk 2 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: I think it works well enough for now with negative x-values

test-plot

File: project.clj (hunk 1 above)
Comment (Eleonore9) [reply project.clj:12]: I get this exception when I remove instaparse: https://gist.github.com/Eleonore9/df42f4a37c37927d4b7015408320043f

File: src/kixi/lapita/plot.clj [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+(defn bar-chart
+  ([ds x-axis y-axis]
+   (bar-chart ds x-axis y-axis {}))
+  ([ds x-axis y-axis {:keys [plot-color plot-width plot-height]}]
+   (let [x-data (sort (wds/subset-ds ds :cols x-axis))
+         x-min (reduce min x-data)
+         x-max (reduce max x-data)
+         y-data (sort (wds/subset-ds ds :cols y-axis))
+         all-data (transform-data-for-plot ds x-data y-data)
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: I'm not sure this is enough to constitute an entirely separate function

File: src/kixi/lapita/plot.clj [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+         bar-width (calc-width-of-bar x-data)
+         color-plot (or plot-color "#3D325A")
+         width-plot (or plot-width 600)
+         height-plot (or plot-height 320)
+         flip-point 0
+         flip-point-height (calc-flip-point flip-point height-plot y-data)
+         flip-line {:values [[(dec x-min) 0] [(inc x-max) 0]]
+                    :attribs {:fill "none" :stroke "#4c4a4a"}
+                    :layout viz/svg-line-plot}
+         plot ((bar-spec all-data 1 bar-width flip-point
```
Comment (acron0) [inline src/kixi/lapita/plot.clj]: Perhaps annotate these magic numbers. `1` is the amount of data series being displayed on the chart and `0` is the index of the series data.

**Review by acron0 — CHANGES_REQUESTED** [review]

File: src/kixi/lapita/plot.clj (hunk 7 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: Good idea!

File: src/kixi/lapita/plot.clj (hunk 6 above)
Comment (Eleonore9) [reply src/kixi/lapita/plot.clj]: OK, I'll change that.

**Review by acron0 — APPROVED** [review]

**Pull request merged**